src/Chainweb/Mempool/InMem.hs

```diff
@@ -611,3 +624,36 @@ getMempoolStats m = do
         <$!> (HashMap.size <$!> readIORef (_inmemPending d))
         <*> (length . _rlRecent <$!> readIORef (_inmemRecentLog d))
         <*> (HashMap.size <$!> readIORef (_inmemBadMap d))
+
+------------------------------------------------------------------------------
+-- | Prune the mempool's pending map and badmap.
+pruneInMem
+    :: forall t . NFData t
+    => MVar (InMemoryMempoolData t)
+    -> IO ()
+pruneInMem lock = do
+    now <- getCurrentTimeIntegral
+    withMVar lock $ \mdata -> pruneInternal mdata now
+
+
+------------------------------------------------------------------------------
+pruneInternal
```

Comment:
It seems like the complexity is linear in the number of pending tx. Is that correct? Could you add and Haddock comment, that states the complexity -- to make sure nobody accidentally uses it in a performance critical path.